### PR TITLE
ref(ui): Extract useOwnerOptions

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This hook may be used to load member and team options for a select component.

### CODE REVIEW

File: static/app/utils/useOwnerOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import type {DetailedTeam, Team} from 'sentry/types';
+import {useMembers} from 'sentry/utils/useMembers';
+import {useTeams} from 'sentry/utils/useTeams';
+import {useTeamsById} from 'sentry/utils/useTeamsById';
+
+interface Options {
+  /**
+   * Props to pass to the leading avatar component
+   */
+  avatarProps?: BaseAvatarProps;
```
[the remark this answers is not in the document]
Comment (evanpurkhiser) [reply static/app/utils/useOwnerOptions.tsx:16]: Yah I can adjust it a bit to move the option generation to a diff hook

[the remark this answers is not in the document]
Comment (evanpurkhiser) [reply static/app/utils/useOwnerOptions.tsx:16]: @ryan953 mind if I get this in as is now actually and I'll get a follow up PR up where it separates the options transformation logic?

I'd like to get the owners filter in asap today too

[the remark this answers is not in the document]
Comment (ryan953) [reply static/app/utils/useOwnerOptions.tsx:16]: yup, sounds good to me!

**Review by ryan953 — APPROVED** [review]

**Review by davidenwang — APPROVED** [review]

**Comment by evanpurkhiser** [comment]
Splitting out over to here actually https://github.com/getsentry/sentry/pull/69348

**Pull request closed without merging**